mmdet/evaluation/metrics/voc_metric.py

```diff
@@ -1,176 +1,170 @@
 # Copyright (c) OpenMMLab. All rights reserved.
 import copy
 import warnings
-from collections import OrderedDict
-from typing import List, Optional, Sequence, Union
+from typing import List, Optional, Sequence, Tuple, Union
 
 import numpy as np
-from mmengine.evaluator import BaseMetric
-from mmengine.logging import MMLogger
+from mmengine.logging import print_log
+from mmeval import VOCMeanAP
+from terminaltables import AsciiTable
 
 from mmdet.registry import METRICS
-from ..functional import eval_map, eval_recalls
 
 
 @METRICS.register_module()
-class VOCMetric(BaseMetric):
-    """Pascal VOC evaluation metric.
+class VOCMetric(VOCMeanAP):
+    """A wrapper of :class:`mmeval.VOCMeanAP`.
+
+    This wrapper implements the `process` method that parses predictions and
+    labels from inputs. This enables ``mmengine.Evaluator`` to handle the data
+    flow of different tasks through a unified interface.
+
+    In addition, this wrapper also implements the ``evaluate`` method that
+    parses metric results and print pretty table of metrics per class.
 
     Args:
-        iou_thrs (float or List[float]): IoU threshold. Defaults to 0.5.
+        iou_thrs (float ｜ List[float]): IoU thresholds. Defaults to 0.5.
         scale_ranges (List[tuple], optional): Scale ranges for evaluating
             mAP. If not specified, all bounding boxes would be included in
             evaluation. Defaults to None.
-        metric (str | list[str]): Metrics to be evaluated. Options are
-            'mAP', 'recall'. If is list, the first setting in the list will
-             be used to evaluate metric.
-        proposal_nums (Sequence[int]): Proposal number used for evaluating
```

Comment:
How to calculate recall now?

Reply:
https://github.com/open-mmlab/mmdetection/pull/9079/files#diff-a3363172750cb51c6471c97aaa3339e00584fe38e7e077f7f9eca47ee9ca497aR13